src/lib/fast_check.py

```diff
@@ -0,0 +1,123 @@
+                },
+                params={
+                    "filter": "state:ENABLED"
+                },
+                timeout=timeout)
+            if response.status != 200:
+                self.logging_context.log(f'Http error: {response.status}, url: {response.url}, reason: {response.reason}')
+                return None
+
+            return await response.json()
+        except Exception as e:
+            self.logging_context.log(f'Unable to get project: {project_id} services list. Error details: {e}')
+            return None
+
+    async def get_dynatrace_token_metadata(self, dynatrace_url: str, dynatrace_api_key: str, timeout: Optional[int] = 2):
+        try:
+            response = await self.session.post(
+                url=urljoin(dynatrace_url, "/api/v1/tokens/lookup"),
+                headers={
+                    "Authorization": f"Api-Token {dynatrace_api_key}",
+                    "Content-Type": "application/json; charset=utf-8"
+                },
+                json={
+                    "token": dynatrace_api_key
+                },
+                timeout=timeout)
+            if response.status != 200:
+                self.logging_context.log(f'Unable to get Dynatrace token metadata: {response.status}, url: {response.url}, reason: {response.reason}')
+                return None
+
+            return await response.json()
+        except Exception as e:
+            self.logging_context.log(f'Unable to get Dynatrace token metadata. Error details: {e}')
+            return None
+
+    async def _check_services(self, project_id):
+        list_services_result = await self.list_services(project_id)
+        service_names = [find_service_name(service['name']) for service in list_services_result['services']] if list_services_result else []
+        if not all(name in service_names for name in required_services):
+            self.logging_context.log(f'Cannot monitor project: \'{project_id}\'. '
+                                     f'Enable required services: {required_services}')
+            return None
+        return service_names
+
+    async def _check_dynatrace(self, project_id):
+        try:
+            dynatrace_url = await fetch_secret(self.session, project_id, self.token, "DYNATRACE_URL")
+            dynatrace_access_key = await fetch_secret(self.session, project_id, self.token, "DYNATRACE_ACCESS_KEY")
+            if not dynatrace_url or not dynatrace_access_key:
+                self.logging_context.log(f'No Dynatrace secrets: DYNATRACE_URL, DYNATRACE_ACCESS_KEY for project: {project_id}.'
+                                         f'Add required secrets to Secret Manager.')
+                return None
+
+            token_metadata = await self.get_dynatrace_token_metadata(dynatrace_url, dynatrace_access_key)
+            if token_metadata['revoked'] or not all(scope in token_metadata['scopes'] for scope in dynatrace_required_token_scopes):
```

Comment:
You could extract `all(scope in token_metadata['scopes'] for scope in dynatrace_required_token_scopes)` into function with descriptive name